src/rdyadvance.c

```diff
@@ -284,7 +344,7 @@ PetscBool RDyFinished(RDy rdy) {
   PetscReal time_in_unit = ConvertTimeFromSeconds(time, rdy->config.time.unit);
   PetscInt  step;
   PetscCall(TSGetStepNumber(rdy->ts, &step));
-  if ((time_in_unit >= rdy->config.time.final_time) || (step >= rdy->config.time.max_step)) {
+  if ((time_in_unit >= rdy->config.time.final_time)) {  // || (step >= rdy->config.time.max_step)) {
```

Comment:
Shouldn't we leave the `max_step` check in here for the fixed-timestep case? We can set `max_step` to `INT_MAX`, which is the maximum (32-bit) value for C integers. This should be appropriate for 64-bit integers, too, because I don't think we want to take 2e9 timesteps in any simulation(!)

Reply:
Aha, I see we have `max_step` set to `INVALID_INT` by default, so maybe this check should be done elsewhere (if it's not already).